src/com/amazon/ionelement/api/AnyElement.kt

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.ionelement.api
+
+import com.amazon.ion.Decimal
+import com.amazon.ion.IntegerSize
+import com.amazon.ion.Timestamp
+import com.amazon.ionelement.api.ElementType.BLOB
+import com.amazon.ionelement.api.ElementType.BOOL
+import com.amazon.ionelement.api.ElementType.CLOB
+import com.amazon.ionelement.api.ElementType.DECIMAL
+import com.amazon.ionelement.api.ElementType.INT
+import com.amazon.ionelement.api.ElementType.LIST
+import com.amazon.ionelement.api.ElementType.NULL
+import com.amazon.ionelement.api.ElementType.SEXP
+import com.amazon.ionelement.api.ElementType.STRING
+import com.amazon.ionelement.api.ElementType.STRUCT
+import com.amazon.ionelement.api.ElementType.SYMBOL
+import com.amazon.ionelement.api.ElementType.TIMESTAMP
+import java.math.BigInteger
+
+/**
+ * Represents an Ion element of an unknown type.
+ *
+ * This type is used whenever the type of Ion data is not guaranteed.  For example, it is returned by all of the
+ * [IonElementLoader] functions and is the data type for children of [ListElement], [SexpElement] and [StructElement].
+ * If the type of an Ion element is known in advance, it is generally better to use the narrowed [IonElement] instead.
+ *
+ * Two categories of methods are present on this type:
+ *
+ * - Value accessors (the `*Value` and `*ValueOrNull` properties)
+ * - Narrowing functions (`as*()` and `as*OrNull()`
+ *
+ * Each of those categories also include nullable and non-nullable variations of those types.
+ *
+ * Use of the accessor functions allow for concise expression of two purposes simultaneously:
+ *
+ * - An expectation of the Ion type of the given element
+ * - An expectation of the nullability of the given element
+ *
+ * If either of these expectations is violated and [IonElectrolyteException] is thrown.  If the given element
+ * has a [IonLocation] in its metadata, it included with the [IonElectrolyteException] which can be used to
+ * generate helpful error messages which point to the specific location of the failure within text or binary Ion data.
+ *
+ * Examples:
+ *
+ * ```
+ * val e: AnyElement = loadSingleElement("1")
+ * // throws if e is not an ion null and an [INT].
```

Comment:
Oh I was referring to clarity of comment, not behavior change. Asking because there seems to be operator precedence to be clarify like `not` & `and`. "throws if is anything but a non-null INT" sounds really clear to me.